### PR TITLE
NETOBSERV-1792: Added update-build script to add downstream labels

### PR DESCRIPTION
**Pull request by OlivierCazade**
## Description

Added update-build script to add downstream labels

## Dependencies

n/a

## Checklist

If you are not familiar with our processes or don't know what to answer in the list below, let us know in a comment: the maintainers will take care of that.

* [X] Is this PR backed with a JIRA ticket? If so, make sure it is written as a title prefix _(in general, PRs affecting the NetObserv/Network Observability product should be backed with a JIRA ticket - especially if they bring user facing changes)._
* [ ] Does this PR require product documentation?
  * [ ] If so, make sure the JIRA epic is labelled with "documentation" and provides a description relevant for doc writers, such as use cases or scenarios. Any required step to activate or configure the feature should be documented there, such as new CRD knobs.
* [ ] Does this PR require a product release notes entry?
  * [ ] If so, fill in "Release Note Text" in the JIRA.
* [ ] Is there anything else the QE team should know before testing? E.g: configuration changes, environment setup, etc.
  * [ ] If so, make sure it is described in the JIRA ticket.
* QE requirements (check 1 from the list):
  * [ ] Standard QE validation, with pre-merge tests unless stated otherwise.
  * [ ] Regression tests only (e.g. refactoring with no user-facing change).
  * [ ] No QE (e.g. trivial change with high reviewer's confidence, or per agreement with the QE team).

### CODE REVIEW

File: hack/update-build.sh [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+EOF
+
+echo "Updating fbc bundle file"
+
+
+cat <<EOF >>"${BUNDLE_CONTAINER_FILE}"
+LABEL com.redhat.component="network-observability-operator-bundle-container"
+LABEL name="network-observability-operator-bundle"
+LABEL io.k8s.display-name="Network Observability Operator Bundle"
+LABEL io.k8s.description="Network Observability Operator Bundle"
```
Comment (jotak) [inline hack/update-build.sh:38]: nit, for consistency with current description

```suggestion
LABEL io.k8s.display-name="Network Observability operator bundle"
LABEL io.k8s.description="Bundle for the Network Observability operator"
```

File: hack/update-build.sh [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+LABEL upstream-vcs-type="git"
+EOF
+
+echo "Updating catalog container file"
+
+cat <<EOF >>"${CATALOG_CONTAINER_FILE}"
+LABEL com.redhat.component="network-observability-operator-catalog-container"
+LABEL name="network-observability-operator-catalog"
+LABEL io.k8s.display-name="Network Observability Operator Catalog"
+LABEL io.k8s.description="Network Observability Operator Catalog"
```
Comment (jotak) [inline hack/update-build.sh:52]:
```suggestion
LABEL io.k8s.display-name="Network Observability operator catalog"
LABEL io.k8s.description="Catalog for the Network Observability operator"
```

**Comment by jotak** [comment]
@OlivierCazade do you have a diff between the konflux-generated bundle and the cpaas-generated one?

I guess we should check that to validate the whole konflux pipeline .. cc @memodi

**Comment by OlivierCazade** [comment]
Here is a diff with the last modifications

<details>
  <summary>diff</summary>

  ```
diff --recursive 1.7-cpaas/bundle/manifests/netobserv-operator.clusterserviceversion.yaml 1.7/bundle/manifests/netobserv-operator.clusterserviceversion.yaml
437,438c437,438
<     containerImage: registry.redhat.io/network-observability/network-observability-rhel9-operator@sha256:9a5596f7cbc1e86627a0c618e44caeb3e3176823e7e43b068798ee26c7c799a8
<     createdAt: '2024-09-04T21:28:32'
---
>     containerImage: ''
>     createdAt: '2024-09-06T12:58:39'
440c440,441
<     operatorframework.io/initialization-resource: '{"apiVersion":"flows.netobserv.io/v1beta2", "kind":"FlowCollector","metadata":{"name":"cluster"},"spec": {}}'
---
>     operatorframework.io/initialization-resource: '{"apiVersion":"flows.netobserv.io/v1beta2",
>       "kind":"FlowCollector","metadata":{"name":"cluster"},"spec": {}}'
446c447,448
<     operators.openshift.io/valid-subscription: '["OpenShift Kubernetes Engine", "OpenShift Container Platform", "OpenShift Platform Plus"]'
---
>     operators.openshift.io/valid-subscription: '["OpenShift Kubernetes Engine", "OpenShift
>       Container Platform", "OpenShift Platform Plus"]'
449c451
<     features.operators.openshift.io/fips-compliant: 'true'
---
>     features.operators.openshift.io/fips-compliant: 'false'
458d459
<     netobserv.io/version-release: v1.7.0-38
471c472,473
<     - description: '`FlowCollector` is the schema for the network flows collection API, which pilots and configures the underlying deployments.'
---
>     - description: '`FlowCollector` is the schema for the network flows collection
>         API, which pilots and configures the underlying deployments.'
476c478,479
<     - description: '`FlowCollector` is the schema for the network flows collection API, which pilots and configures the underlying deployments.'
---
>     - description: '`FlowCollector` is the schema for the network flows collection
>         API, which pilots and configures the underlying deployments.'
563c566,568
<       - description: of the component that receives the flows from the agent, enriches them, generates metrics, and forwards them to the Loki persistence layer and/or any available exporter.
---
>       - description: of the component that receives the flows from the agent, enriches
>           them, generates metrics, and forwards them to the Loki persistence layer
>           and/or any available exporter.
830c835,836
<       - description: Namespace where console plugin and flowlogs-pipeline have been deployed.
---
>       - description: Namespace where console plugin and flowlogs-pipeline have been
>           deployed.
841c847,848
<     - description: '`FlowMetric` is the schema for the custom metrics API, which allows to generate more metrics out of flow logs. You can find examples here: https://github.com/netobserv/network-observability-operator/tree/main/config/samples/flowmetrics'
---
>     - description: '`FlowMetric` is the schema for the custom metrics API, which allows
>         to generate more metrics out of flow logs. You can find examples here: https://github.com/netobserv/network-observability-operator/tree/main/config/samples/flowmetrics'
924c931,932
<   - base64data: PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0idXRmLTgiPz4KPCEtLSBHZW5lcmF0b3I6IEFkb2JlIElsbHVzdHJhdG9yIDI2LjAuMiwgU1ZHIEV4cG9ydCBQbHVnLUluIC4gU1ZHIFZlcnNpb246IDYuMDAgQnVpbGQgMCkgIC0tPgo8c3ZnIHZlcnNpb249IjEuMSIgaWQ9IkxheWVyXzEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiIHg9IjBweCIgeT0iMHB4IgoJIHZpZXdCb3g9IjAgMCAxMDAgMTAwIiBzdHlsZT0iZW5hYmxlLWJhY2tncm91bmQ6bmV3IDAgMCAxMDAgMTAwOyIgeG1sOnNwYWNlPSJwcmVzZXJ2ZSI+CjxzdHlsZSB0eXBlPSJ0ZXh0L2NzcyI+Cgkuc3Qwe2ZpbGw6dXJsKCNTVkdJRF8xXyk7fQoJLnN0MXtmaWxsOiNGRkZGRkY7fQoJLnN0MntvcGFjaXR5OjAuNjt9Cgkuc3Qze29wYWNpdHk6MC41O30KCS5zdDR7b3BhY2l0eTowLjQ7fQo8L3N0eWxlPgo8Zz4KCTxnPgoJCTxnPgoJCQk8cmFkaWFsR3JhZGllbnQgaWQ9IlNWR0lEXzFfIiBjeD0iMTQuNzc1OCIgY3k9Ii0yLjk3NzEiIHI9IjkxLjYyNyIgZ3JhZGllbnRVbml0cz0idXNlclNwYWNlT25Vc2UiPgoJCQkJPHN0b3AgIG9mZnNldD0iMCIgc3R5bGU9InN0b3AtY29sb3I6IzNDM0ZBNiIvPgoJCQkJPHN0b3AgIG9mZnNldD0iMSIgc3R5bGU9InN0b3AtY29sb3I6IzNCMDM0MCIvPgoJCQk8L3JhZGlhbEdyYWRpZW50PgoJCQk8cGF0aCBjbGFzcz0ic3QwIiBkPSJNNTAsOTljLTEzLjMsMC0yNS40LTUuMy0zNC4yLTEzLjlDNi43LDc2LjIsMSw2My43LDEsNTBDMSwyMi45LDIyLjksMSw1MCwxYzEzLjcsMCwyNi4yLDUuNywzNS4xLDE0LjgKCQkJCUM5My43LDI0LjYsOTksMzYuNyw5OSw1MEM5OSw3Ny4xLDc3LjEsOTksNTAsOTl6Ii8+CgkJPC9nPgoJCTxnPgoJCQk8Y2lyY2xlIGNsYXNzPSJzdDEiIGN4PSIzNy41IiBjeT0iODEuOSIgcj0iNSIvPgoJCTwvZz4KCQk8cGF0aCBjbGFzcz0ic3QxIiBkPSJNNDguNiw5MS45bDE4LjgtNDMuM2MtMi41LTAuMS01LTAuNy03LjItMkwzMy4yLDY4LjJsMS40LTEuOGwyMC0yNS4xYy0xLjUtMi40LTIuMy01LjEtMi4zLTcuOUw5LDUyLjIKCQkJbDQ3LjYtMjkuOWwwLDBjMC4xLTAuMSwwLjItMC4yLDAuMi0wLjJjNi4xLTYuMSwxNS45LTYuMSwyMiwwbDAuMSwwLjFjNiw2LjEsNiwxNS45LTAuMSwyMS45Yy0wLjEsMC4xLTAuMiwwLjItMC4yLDAuMmwwLDAKCQkJTDQ4LjYsOTEuOXoiLz4KCQk8ZyBjbGFzcz0ic3QyIj4KCQkJPGNpcmNsZSBjbGFzcz0ic3QxIiBjeD0iNTAuMyIgY3k9IjE0LjciIHI9IjMuMSIvPgoJCTwvZz4KCQk8ZyBjbGFzcz0ic3QzIj4KCQkJPGNpcmNsZSBjbGFzcz0ic3QxIiBjeD0iMjcuNyIgY3k9IjU4IiByPSIxLjciLz4KCQk8L2c+CgkJPGc+CgkJCTxjaXJjbGUgY2xhc3M9InN0MSIgY3g9Ijc3LjQiIGN5PSI2OS4zIiByPSIxLjciLz4KCQk8L2c+CgkJPGc+CgkJCTxjaXJjbGUgY2xhc3M9InN0MSIgY3g9IjE2LjMiIGN5PSIzNi42IiByPSIxLjciLz4KCQk8L2c+CgkJPGcgY2xhc3M9InN0NCI+CgkJCTxjaXJjbGUgY2xhc3M9InN0MSIgY3g9IjYzLjciIGN5PSI4NS45IiByPSIyLjIiLz4KCQk8L2c+CgkJPGc+CgkJCTxjaXJjbGUgY2xhc3M9InN0MSIgY3g9IjI5LjQiIGN5PSIxOS42IiByPSI0LjgiLz4KCQk8L2c+CgkJPGcgY2xhc3M9InN0MyI+CgkJCTxjaXJjbGUgY2xhc3M9InN0MSIgY3g9Ijg4IiBjeT0iNTAiIHI9IjQuOCIvPgoJCTwvZz4KCTwvZz4KPC9nPgo8L3N2Zz4K
---
>   - base64data:
>       PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0idXRmLTgiPz4KPCEtLSBHZW5lcmF0b3I6IEFkb2JlIElsbHVzdHJhdG9yIDI2LjAuMiwgU1ZHIEV4cG9ydCBQbHVnLUluIC4gU1ZHIFZlcnNpb246IDYuMDAgQnVpbGQgMCkgIC0tPgo8c3ZnIHZlcnNpb249IjEuMSIgaWQ9IkxheWVyXzEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiIHg9IjBweCIgeT0iMHB4IgoJIHZpZXdCb3g9IjAgMCAxMDAgMTAwIiBzdHlsZT0iZW5hYmxlLWJhY2tncm91bmQ6bmV3IDAgMCAxMDAgMTAwOyIgeG1sOnNwYWNlPSJwcmVzZXJ2ZSI+CjxzdHlsZSB0eXBlPSJ0ZXh0L2NzcyI+Cgkuc3Qwe2ZpbGw6dXJsKCNTVkdJRF8xXyk7fQoJLnN0MXtmaWxsOiNGRkZGRkY7fQoJLnN0MntvcGFjaXR5OjAuNjt9Cgkuc3Qze29wYWNpdHk6MC41O30KCS5zdDR7b3BhY2l0eTowLjQ7fQo8L3N0eWxlPgo8Zz4KCTxnPgoJCTxnPgoJCQk8cmFkaWFsR3JhZGllbnQgaWQ9IlNWR0lEXzFfIiBjeD0iMTQuNzc1OCIgY3k9Ii0yLjk3NzEiIHI9IjkxLjYyNyIgZ3JhZGllbnRVbml0cz0idXNlclNwYWNlT25Vc2UiPgoJCQkJPHN0b3AgIG9mZnNldD0iMCIgc3R5bGU9InN0b3AtY29sb3I6IzNDM0ZBNiIvPgoJCQkJPHN0b3AgIG9mZnNldD0iMSIgc3R5bGU9InN0b3AtY29sb3I6IzNCMDM0MCIvPgoJCQk8L3JhZGlhbEdyYWRpZW50PgoJCQk8cGF0aCBjbGFzcz0ic3QwIiBkPSJNNTAsOTljLTEzLjMsMC0yNS40LTUuMy0zNC4yLTEzLjlDNi43LDc2LjIsMSw2My43LDEsNTBDMSwyMi45LDIyLjksMSw1MCwxYzEzLjcsMCwyNi4yLDUuNywzNS4xLDE0LjgKCQkJCUM5My43LDI0LjYsOTksMzYuNyw5OSw1MEM5OSw3Ny4xLDc3LjEsOTksNTAsOTl6Ii8+CgkJPC9nPgoJCTxnPgoJCQk8Y2lyY2xlIGNsYXNzPSJzdDEiIGN4PSIzNy41IiBjeT0iODEuOSIgcj0iNSIvPgoJCTwvZz4KCQk8cGF0aCBjbGFzcz0ic3QxIiBkPSJNNDguNiw5MS45bDE4LjgtNDMuM2MtMi41LTAuMS01LTAuNy03LjItMkwzMy4yLDY4LjJsMS40LTEuOGwyMC0yNS4xYy0xLjUtMi40LTIuMy01LjEtMi4zLTcuOUw5LDUyLjIKCQkJbDQ3LjYtMjkuOWwwLDBjMC4xLTAuMSwwLjItMC4yLDAuMi0wLjJjNi4xLTYuMSwxNS45LTYuMSwyMiwwbDAuMSwwLjFjNiw2LjEsNiwxNS45LTAuMSwyMS45Yy0wLjEsMC4xLTAuMiwwLjItMC4yLDAuMmwwLDAKCQkJTDQ4LjYsOTEuOXoiLz4KCQk8ZyBjbGFzcz0ic3QyIj4KCQkJPGNpcmNsZSBjbGFzcz0ic3QxIiBjeD0iNTAuMyIgY3k9IjE0LjciIHI9IjMuMSIvPgoJCTwvZz4KCQk8ZyBjbGFzcz0ic3QzIj4KCQkJPGNpcmNsZSBjbGFzcz0ic3QxIiBjeD0iMjcuNyIgY3k9IjU4IiByPSIxLjciLz4KCQk8L2c+CgkJPGc+CgkJCTxjaXJjbGUgY2xhc3M9InN0MSIgY3g9Ijc3LjQiIGN5PSI2OS4zIiByPSIxLjciLz4KCQk8L2c+CgkJPGc+CgkJCTxjaXJjbGUgY2xhc3M9InN0MSIgY3g9IjE2LjMiIGN5PSIzNi42IiByPSIxLjciLz4KCQk8L2c+CgkJPGcgY2xhc3M9InN0NCI+CgkJCTxjaXJjbGUgY2xhc3M9InN0MSIgY3g9IjYzLjciIGN5PSI4NS45IiByPSIyLjIiLz4KCQk8L2c+CgkJPGc+CgkJCTxjaXJjbGUgY2xhc3M9InN0MSIgY3g9IjI5LjQiIGN5PSIxOS42IiByPSI0LjgiLz4KCQk8L2c+CgkJPGcgY2xhc3M9InN0MyI+CgkJCTxjaXJjbGUgY2xhc3M9InN0MSIgY3g9Ijg4IiBjeT0iNTAiIHI9IjQuOCIvPgoJCTwvZz4KCTwvZz4KPC9nPgo8L3N2Zz4K
1170c1178
<                 bundleCreatedAt: '2024-09-04T21:28:32'
---
>                 bundleCreatedAt: '2024-09-06T12:58:39'
1186c1194,1195
<                   value: registry.redhat.io/network-observability/network-observability-ebpf-agent-rhel9@sha256:226ef86f44b514f5f332b648245be0b1e4aa4c756631774f4a03014b5394d6f9
---
>                   value:
>                     registry.redhat.io/network-observability/network-observability-ebpf-agent-rhel9@v1.7.0
1188c1197,1198
<                   value: registry.redhat.io/network-observability/network-observability-flowlogs-pipeline-rhel9@sha256:c462cc4dc78ce883427c662bf3996a87683dc02c8d94a1a4193f78077878ea6a
---
>                   value:
>                     registry.redhat.io/network-observability/network-observability-flowlogs-pipeline-rhel9@v1.7.0
1190c1200,1201
<                   value: registry.redhat.io/network-observability/network-observability-console-plugin-rhel9@sha256:3c3e09dc489b59bb8ad5e1b5b80c81591b223187eef18d61a6884ca42729e3d2
---
>                   value:
>                     registry.redhat.io/network-observability/network-observability-console-plugin-rhel9@v1.7.0
1194c1205,1206
<                 image: registry.redhat.io/network-observability/network-observability-rhel9-operator@sha256:9a5596f7cbc1e86627a0c618e44caeb3e3176823e7e43b068798ee26c7c799a8
---
>                 image:
>                   registry.redhat.io/network-observability/network-observability-rhel9-operator@v1.7.0
1237c1249
<                 image: registry.redhat.io/openshift4/ose-kube-rbac-proxy@sha256:f6c37678f1eb3279e603f63d2a821b72394c52d25c2ed5058dc29d4caa15d504
---
>                 image: gcr.io/kubebuilder/kube-rbac-proxy:v0.15.0
1408,1420d1419
<   relatedImages:
<   - name: network-observability-rhel9-operator-9a5596f7cbc1e86627a0c618e44caeb3e3176823e7e43b068798ee26c7c799a8-annotation
<     image: registry.redhat.io/network-observability/network-observability-rhel9-operator@sha256:9a5596f7cbc1e86627a0c618e44caeb3e3176823e7e43b068798ee26c7c799a8
<   - name: manager
<     image: registry.redhat.io/network-observability/network-observability-rhel9-operator@sha256:9a5596f7cbc1e86627a0c618e44caeb3e3176823e7e43b068798ee26c7c799a8
<   - name: kube-rbac-proxy
<     image: registry.redhat.io/openshift4/ose-kube-rbac-proxy@sha256:f6c37678f1eb3279e603f63d2a821b72394c52d25c2ed5058dc29d4caa15d504
<   - name: ebpf_agent
<     image: registry.redhat.io/network-observability/network-observability-ebpf-agent-rhel9@sha256:226ef86f44b514f5f332b648245be0b1e4aa4c756631774f4a03014b5394d6f9
<   - name: flowlogs_pipeline
<     image: registry.redhat.io/network-observability/network-observability-flowlogs-pipeline-rhel9@sha256:c462cc4dc78ce883427c662bf3996a87683dc02c8d94a1a4193f78077878ea6a
<   - name: console_plugin
<     image: registry.redhat.io/network-observability/network-observability-console-plugin-rhel9@sha256:3c3e09dc489b59bb8ad5e1b5b80c81591b223187eef18d61a6884ca42729e3d2
diff --recursive 1.7-cpaas/bundle/metadata/annotations.yaml 1.7/bundle/metadata/annotations.yaml
16d15
<   com.redhat.openshift.versions: v4.13
Only in 1.7-cpaas/bundle: root
Only in 1.7/bundle: tests
  ```
</details>

File: hack/patch_csv.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+flp_image ='registry.redhat.io/network-observability/network-observability-flowlogs-pipeline-rhel9@v{}'.format(version)
+console_image = 'registry.redhat.io/network-observability/network-observability-console-plugin-rhel9@v{}'.format(version)
+
+csv['metadata']['annotations']['operators.openshift.io/valid-subscription'] = '["OpenShift Kubernetes Engine", "OpenShift Container Platform", "OpenShift Platform Plus"]'
+csv['metadata']['annotations']['operatorframework.io/cluster-monitoring'] = 'true'
+csv['metadata']['name'] = 'network-observability-operator.v{}'.format(version)
+csv['metadata']['annotations']['createdAt'] = created_at
+csv['metadata']['annotations']['containerImage'] = os.getenv('OPERATOR_PULLSPEC', '')
+csv['metadata']['annotations']['features.operators.openshift.io/disconnected'] = 'true'
+csv['metadata']['annotations']['features.operators.openshift.io/fips-compliant'] = 'false'
```
Comment (jotak) [inline hack/patch_csv.py:40]: FIPS should be true
```suggestion
csv['metadata']['annotations']['features.operators.openshift.io/fips-compliant'] = 'true'
```

File: hack/update-build.sh [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+#!/usr/bin/env bash
+
+echo "Updating container file"
+
+: "${COMMIT:=$(git rev-list --abbrev-commit --tags --max-count=1)}"
+: "${CONTAINER_FILE:=./Dockerfile}"
+: "${BUNDLE_CONTAINER_FILE:=./bundle.Dockerfile}"
+: "${CATALOG_CONTAINER_FILE:=./catalog.Dockerfile}"
+: "${TARGET_VERSION:=1.7.0}"
+: "${REPLACE_VERSION:=1.6.0}"
```
Comment (jotak) [inline hack/update-build.sh:11]: Current latest is 1.6.1

```suggestion
: "${REPLACE_VERSION:=1.6.1}"
```

**Pull request merged**